tool/starter

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env perl
+use 5.016;
+use warnings;
+use autodie;
+
+use File::Path qw/mkpath/;
+
+my $plug = shift or die 'plugin name required';
+my $pkg_base = lc($plug =~ s/[^a-zA-Z0-9]//gr);
+my $struct = ucfirst $pkg_base;
+
+my $main_go = qq{package main
+
+import "github.com/mackerelio/mackerel-agent-plugins/mackerel-plugin-$plug/lib"
+
+func main() {
+	mp$pkg_base.Do()
+}
+};
+
+my $readme = qq{mackerel-plugin-$plug
+=====================
+
+$plug custom metrics plugin for mackerel.io agent.
+
+## Synopsis
+
+```shell
+mackerel-plugin-$plug
+```
+
+## Example of mackerel-agent.conf
+
+```
+[plugin.metrics.$plug]
+command = "/path/to/mackerel-plugin-$plug"
+```
+};
+
+my $src = qq|package mp$pkg_base
+
+import (
+	"flag"
+
+	mp "github.com/mackerelio/go-mackerel-plugin"
+)
+
+// ${struct}Plugin mackerel plugin
+type ${struct}Plugin struct {
+	prefix string
+}
+
+// MetricKeyPrefix interface for PluginWithPrefix
+func (p ${struct}Plugin) MetricKeyPrefix() string {
+	if p.prefix == "" {
+		p.prefix = "$pkg_base"
```

Comment:
The `MetricKeyPrefix` should return a default string, even if not explicitly specified. Because, the Plugin types should provide default behavior independent of flags and passing flags is to change the default behavior. It is preferable for document generation.